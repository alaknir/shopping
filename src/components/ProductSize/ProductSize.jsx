import React from "react";
import PropTypes from "prop-types";

const CustomRadio = ({ productId, size }) => (
  <label for={`${productId}_${size}`}>
    {size}
    <input name={productId} id={`${productId}_${size}`} type="radio" />
  </label>
);

CustomRadio.propTypes = {
  productId: PropTypes.string.isRequired,
  sizeList: PropTypes.string.isRequired
};

const ProductSize = ({ productId, sizeList }) => [
  sizeList.map(size => <CustomRadio productId={productId} size={size} />)
];

ProductSize.propTypes = {
  productId: PropTypes.string.isRequired,
  sizeList: PropTypes.shape([])
};

ProductSize.defaultProps = {
  sizeList: []
};

export default ProductSize;
