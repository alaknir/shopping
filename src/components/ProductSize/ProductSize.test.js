import React from "react";
import { mount } from "enzyme";

import ProductSize from "./ProductSize";

describe("ProductSize ", () => {
  it("Should render properly", () => {
    const component = (
      <ProductSize productId="1234" size={["XS", "S", "M", "L", "XL"]} />
    );
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
