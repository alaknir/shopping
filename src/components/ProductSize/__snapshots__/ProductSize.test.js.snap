// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSize  Should render properly 1`] = `
<ProductSize
  productId="1234"
  size={
    Array [
      "XS",
      "S",
      "M",
      "L",
      "XL",
    ]
  }
  sizeList={Array []}
/>
`;
